packages/client/src/schedule-client.ts

```diff
@@ -0,0 +1,910 @@
+import { status as grpcStatus } from '@grpc/grpc-js';
+import {
+  compileRetryPolicy,
+  DataConverter,
+  LoadedDataConverter,
+  mapFromPayloads,
+  mapToPayloads,
+  searchAttributePayloadConverter,
+  SearchAttributes,
+  Workflow,
+} from '@temporalio/common';
+import { composeInterceptors, Headers } from '@temporalio/common/lib/interceptors';
+import {
+  encodeMapToPayloads,
+  loadDataConverter,
+  isLoadedDataConverter,
+  encodeToPayloads,
+  filterNullAndUndefined,
+  decodeMapFromPayloads,
+  decodeArrayFromPayloads,
+} from '@temporalio/common/lib/internal-non-workflow';
+import os from 'os';
+import { Connection } from './connection';
+import { CreateScheduleInput, ScheduleClientCallsInterceptor, ScheduleClientInterceptors } from './interceptors';
+import { ConnectionLike, Metadata, WorkflowService } from './types';
+import { v4 as uuid4 } from 'uuid';
+import { isServerErrorResponse, ServiceError } from './errors';
+import {
+  Backfill,
+  CalendarSpec,
+  CalendarSpecDescription,
+  CompiledScheduleOptions,
+  CompiledUpdatedSchedule,
+  compileScheduleOptions,
+  compileUpdatedScheduleOptions,
+  dayOfWeekNameToNumber,
+  dayOfWeekNumberToName,
+  IntervalSpecDescription,
+  ListScheduleEntry,
+  ListScheduleOptions,
+  monthNameToNumber,
+  monthNumberToName,
+  Range,
+  Schedule,
+  ScheduleActionType,
+  ScheduleOptions,
+  ScheduleOverlapPolicy,
+  UpdatedSchedule,
+  UpdateScheduleOptions,
+} from './schedule-types';
+import { Replace } from '@temporalio/common/lib/type-helpers';
+import { temporal } from '@temporalio/proto';
+import {
+  msOptionalToTs,
+  msToTs,
+  optionalDateToTs,
+  optionalTsToDate,
+  optionalTsToMs,
+  tsToDate,
+} from '@temporalio/common/lib/time';
+
+type CreateScheduleRequest = temporal.api.workflowservice.v1.ICreateScheduleRequest;
+// type CreateScheduleResponse = temporal.api.workflowservice.v1.ICreateScheduleResponse;
+type DescribeScheduleRequest = temporal.api.workflowservice.v1.IDescribeScheduleRequest;
+type DescribeScheduleResponse = temporal.api.workflowservice.v1.IDescribeScheduleResponse;
+type DeleteScheduleRequest = temporal.api.workflowservice.v1.IDeleteScheduleRequest;
+type DeleteScheduleResponse = temporal.api.workflowservice.v1.IDeleteScheduleResponse;
+type UpdateScheduleRequest = temporal.api.workflowservice.v1.IUpdateScheduleRequest;
+type UpdateScheduleResponse = temporal.api.workflowservice.v1.IUpdateScheduleResponse;
+type PatchScheduleRequest = temporal.api.workflowservice.v1.IPatchScheduleRequest;
+type PatchScheduleResponse = temporal.api.workflowservice.v1.IPatchScheduleResponse;
+type ListSchedulesRequest = temporal.api.workflowservice.v1.IListSchedulesRequest;
+type ListSchedulesResponse = temporal.api.workflowservice.v1.IListSchedulesResponse;
+
+// FIXME: Determine if we want to support these
+type ListScheduleMatchingTimesRequest = temporal.api.workflowservice.v1.IListScheduleMatchingTimesRequest;
```

Comment:
@lorensr High level API for this request were left out of the proposal. What's the status? Is that in scope for this PR?

Reply:
@dnr has list matching been finalized yet?

<img width="1459" alt="image" src="https://user-images.githubusercontent.com/251288/198364047-ade0ecc4-b75e-4ef9-81be-01ecca3d2640.png">

Reply:
The first API is implemented and works. The problem is that I just don't think it's particularly useful at all. The UI ended up not needing it. So I was considering removing it, but haven't done that yet. Do you (any of you) think it would be useful?

The second (with a separate spec) could be more useful, but isn't implemented yet. It would probably be more useful to the UI than to SDK clients.